src/SerialLoops/Editors/ScriptEditor.cs

```diff
@@ -807,46 +984,405 @@ private void ChibiDropDown_SelectedKeyChanged(object sender, EventArgs e)
+                .Objects[dropDown.Command.Index].Parameters[dropDown.ParameterIndex] =
+                _script.Event.LabelsSection.Objects.First(l => ((ScriptSectionScriptParameter)dropDown.Command.Parameters[dropDown.ParameterIndex]).Section.Name.Replace("/", "") == dropDown.SelectedKey).Id;
+            UpdateTabTitle(false);
+        }
+        private void SfxModeDropDown_SelectedKeyChanged(object sender, EventArgs e)
+        {
+            ScriptCommandDropDown dropDown = (ScriptCommandDropDown)sender;
+            _log.Log($"Attempting to modify parameter {dropDown.ParameterIndex} to SFX mode {dropDown.SelectedKey} in {dropDown.Command.Index} in file {_script.Name}...");
+            ((SfxModeScriptParameter)dropDown.Command.Parameters[dropDown.ParameterIndex]).Mode =
+                Enum.Parse<SfxModeScriptParameter.SfxMode>(dropDown.SelectedKey);
+            _script.Event.ScriptSections[_script.Event.ScriptSections.IndexOf(dropDown.Command.Section)]
+                .Objects[dropDown.Command.Index].Parameters[dropDown.ParameterIndex] =
+                (short)Enum.Parse<SfxModeScriptParameter.SfxMode>(dropDown.SelectedKey);
+            UpdateTabTitle(false);
+        }
+        private void ShortTextBox_TextChanged(object sender, EventArgs e)
+        {
+            ScriptCommandTextBox textBox = (ScriptCommandTextBox)sender;
+            _log.Log($"Attempting to modify parameter {textBox.ParameterIndex} to SFX mode {textBox.Text} in {textBox.Command.Index} in file {_script.Name}...");
+            if (short.TryParse(textBox.Text, out short value))
+            {
+                ((ShortScriptParameter)textBox.Command.Parameters[textBox.ParameterIndex]).Value = value;
+                _script.Event.ScriptSections[_script.Event.ScriptSections.IndexOf(textBox.Command.Section)]
+                    .Objects[textBox.Command.Index].Parameters[textBox.ParameterIndex] = value;
+                UpdateTabTitle(false);
+            }
+        }
+        private void SpriteEntranceDropDown_SelectedKeyChanged(object sender, EventArgs e)
+        {
+            ScriptCommandDropDown dropDown = (ScriptCommandDropDown)sender;
+            _log.Log($"Attempting to modify parameter {dropDown.ParameterIndex} to sprite entrance {dropDown.SelectedKey} in {dropDown.Command.Index} in file {_script.Name}...");
+            ((SpriteEntranceScriptParameter)dropDown.Command.Parameters[dropDown.ParameterIndex]).EntranceTransition =
+                Enum.Parse<SpriteEntranceScriptParameter.SpriteEntranceTransition>(dropDown.SelectedKey);
+            _script.Event.ScriptSections[_script.Event.ScriptSections.IndexOf(dropDown.Command.Section)]
+                .Objects[dropDown.Command.Index].Parameters[dropDown.ParameterIndex] =
+                (short)Enum.Parse<SpriteEntranceScriptParameter.SpriteEntranceTransition>(dropDown.SelectedKey);
+            UpdateTabTitle(false);
```

Comment:
This needs to update the preview too, as changing the transition from left-right will cause character sprites to be re-positioned (though this isn't the case with the exit transition dropdown as the effect of that is only apparent to the subsequent command).
```suggestion
            UpdateTabTitle(false);
            Application.Instance.Invoke(() => UpdatePreview());
```